### ISSUE TITLE
Extract messages from custom hook

### ISSUE THREAD

**Issue by Fxlr8** [formatjs/formatjs#2368]
Hi, I created a simple `useFormatMessage` hook. It looks like this:

```ts
export const useFormatMessage = () => {
  const { formatMessage } = useContext(IntlContext);

  return formatMessage;
};
```

Then I import it and use it in my components. Is it possible to to extract messages from it? There is a `additionalComponentNames` option in the plugin config. Can I somehow add a custom function there?

**Comment by longlho** [comment]
yes that's technically possible. We'll keep this open to see how popular this request is.
